api/service.py

```
@@ -634,6 +649,93 @@ async def get_anomaly_history(
+            name=key.name,
+            permissions=key.permissions,
+            created_at=key.created_at,
+            expires_at=key.expires_at,
+            last_used=key.last_used
+        )
+        for key in api_keys
+    ]
+
+
+@app.delete("/api/v1/auth/apikeys/{key_id}")
+async def revoke_api_key(key_id: str, current_user: User = Depends(get_current_user)):
```

Ruff (0.14.10)

Undefined name `User`

(F821)